roles/transmissionvpn/tasks/main.yml

```diff
@@ -0,0 +1,84 @@
+#########################################################################
+# Title:            Cloudbox Community: transmissionvpn                     #
+# Author(s):        Tarek                                               #
+# URL:              https://github.com/Cloudbox/Community               #
+# Docker Image(s):  haugene/transmission-openvpn                        #
+# --                                                                    #
+#         Part of the Cloudbox project: https://cloudbox.works          #
+#########################################################################
+#                   GNU General Public License v3.0                     #
+#########################################################################
+---
+- name: "Setting CloudFlare DNS Record"
+  include_role:
+    name: cloudflare-dns
+  vars:
+    subdomain: transmission
+  when: cloudflare_enabled
+
+- name: Stop and remove any existing container
+  docker_container:
+    name: transmission
+    state: absent
+
+- name: Create transmissionvpn directories
+  file: "path={{ item }} state=directory mode=0775 owner={{ user.name }} group={{ user.name }}"
+  with_items:
+    - /opt/transmissionvpn
+
+- name: Create new downloads directories
+  file: "path={{ item }} state=directory mode=0775 owner={{ user.name }} group={{ user.name }} recurse=yes"
+  with_items:
+    - "{{ downloads.torrents }}"
+    - "{{ downloads.torrents }}/transmissionvpn"
+    - "{{ downloads.torrents }}/transmissionvpn/completed"
+    - "{{ downloads.torrents }}/transmissionvpn/incoming"
+    - "{{ downloads.torrents }}/transmissionvpn/watched"
+    - "{{ downloads.torrents }}/transmissionvpn/torrents"
+    - "{{ downloads.torrents }}/transmissionvpn/vpn"
+  when: (torrents_downloads_path is defined)|default(false)
+
+- name: Set default_volumes variable
+  set_fact:
+    default_volumes:
+      - "/etc/localtime:/etc/localtime:ro"
+      - "/opt/transmissionvpn/vpn:/etc/openvpn/custom"
+      - "/data:/data"
+
+- name: Create and start container
+  docker_container:
+    name: transmissionvpn
+    image: haugene/transmission-openvpn:latest-alpine
+    pull: yes
+    published_ports:
+      - "9091:9091"
```

Comment:
Since it’s going though reverse proxy, doesn’t need this to be open, right? If so, Could either remove published poets altogether rather specify localhost IP access only.

Reply:
I'm not sure how nginx can work without this port, put if it can work then we can delete it , can you advise me ?

Reply:
443 will go to 9091. That’s what virtual port does.

Reply:
not in all access, when connecting between apps its needed